RobotController/robot_brain.py

```diff
@@ -61,23 +65,35 @@ def _check_next_evaluation(self, force=False):
         Check if is a moment for a new evaluation and starts a new one
         :param force: forces the new evaluation to start
         """
+        if self.mating_client and self.mating_client.availability:
+            self.mating_client.server.join()
+            self.mating_client.client.join()
+            self.mating_client = None
```

Comment:
those two joins will make the robot stop to move for a while, are you sure they join quickly?